Main.cpp

```diff
@@ -0,0 +1,133 @@
+/* 
+ @file      Main.cpp
+ @brief     creates geometry and materials to run a Monte Carlo simulation
+ @author    Luke Eure
+ @date      January 9 2016
+*/
+
+#include <iostream>
+#include <vector>
+#include <time.h>
+#include <math.h>
+#include <stdlib.h>
+
+#include "Material.h"
+#include "Boundaries.h"
+#include "Distributions.h"
+#include "Tally.h"
+#include "Mesh.h"
+#include "Monte_carlo.h"
+#include "Plotter.h"
+#include "Neutron.h"
+
+int main() {
+    
+    srand(time(NULL));
+
+    // create geometry
+    Boundaries test_boundary;
+    test_boundary.setSurfaceCoord(0, 0, -2.0);
+    test_boundary.setSurfaceCoord(0, 1, 2.0);
+    test_boundary.setSurfaceCoord(1, 0, -2.0);
+    test_boundary.setSurfaceCoord(1, 1, 2.0);
+    test_boundary.setSurfaceCoord(2, 0, -2.0);
+    test_boundary.setSurfaceCoord(2, 1, 2.0);
+    test_boundary.setSurfaceType(0, 0, VACUUM);
+    test_boundary.setSurfaceType(0, 1, VACUUM);
+    test_boundary.setSurfaceType(1, 0, VACUUM);
+    test_boundary.setSurfaceType(1, 1, VACUUM);
+    test_boundary.setSurfaceType(2, 0, VACUUM);
+    test_boundary.setSurfaceType(2, 1, VACUUM);
+
+    const int num_groups = 2;
+
+    // fuel cross sections
```

Comment:
Cross sections should be wrapped up in a class.  I personally like one object per Isotope in the simulation.  Here is my own personal `Isotope` class that I'm using in a simple MC code of mine to give you an idea of what can be done here.  This is the base class... derived classes need to override `sigmaA()` and `sigmaS()` in order provide the microscopic XS.  Those can either be constant (I have a `ConstantXSIsotope`) or based on E (like groups).

```C++
/**
 * Provides XS for an Isotope
 */
class Isotope
{
public:
  enum Reaction
  {
    ABSORB,
    SCATTER
  };

  /**
   * Constructor
   * @param N Number density
   */
  Isotope(double N):
      _N(N)
  {}

  /**
   * Micrscopic Absorption
   * @param E Energy
   */
  virtual double sigmaA(double E) = 0;

  /**
   * Micrscopic Scattering
   * @param E Energy
   */
  virtual double sigmaS(double E) = 0;

  /**
   * Micrscopic Total
   * @param E Energy
   */
  virtual double sigmaT(double E)
  {
    return sigmaA(E) + sigmaS(E);
  }

  /**
   * Micrscopic Absorption
   * @param E Energy
   */
  virtual double SigmaA(double E)
  {
    return _N*1e-24*sigmaA(E);
  }

  /**
   * Micrscopic Scattering
   * @param E Energy
   */
  virtual double SigmaS(double E)
  {
    return _N*1e-24*sigmaS(E);
  }

  /**
   * Micrscopic Total
   * @param E Energy
   */
  virtual double SigmaT(double E)
  {
    return _N*1e-24*sigmaT(E);
  }

  /**
   * Sample reaction
   * @param xi A random value between 0 and 1
   * @parma E The energy
   * @return The type of reaction
   */
  virtual Reaction sampleReaction(double E, double xi)
  {
    double sigma_t = sigmaT(E);

    double sigma_a = sigmaA(E);

    if (xi <= sigma_a / sigma_t)
      return ABSORB;
    else
      return SCATTER;
  }

protected:
  /// Number density
  double _N;
};
```